### ISSUE TITLE
Label icons not correctly shown on Safari and Firefox

### ISSUE THREAD

**Issue by Soben713** [ProjectSidewalk/SidewalkWebpage#316]
Here is the video: https://youtu.be/c6K1Wz8KNwo

The first browser is Safari and then I switch to Firefox. Sorry that it's not clear in the video. Also, the label colors strangely was changed in the screen recording software (e.g., it is purple for the missing curb ramp) just ignore that, it didn't happen in the actual experiment.

**Comment by jonfroehlich** [comment]
The FireFox bug was noted here: https://github.com/ProjectSidewalk/SidewalkWebpage/issues/177.

**Comment by jonfroehlich** [comment]
So far, 63.55% of users are using Chrome, then 13.47% Safari mobile (I believe that is Safari (in-app), which refers to users visiting our site from Facebook or Twitter, etc. on their IOS devices), then 11% Safari. Note: nearly 20% of our users are Safari or Firefox so it seems like testing on these platforms and getting them to work is important. I'm assigning this to @Soben713. 

![image](https://cloud.githubusercontent.com/assets/1621749/18333407/cdaabf4c-753c-11e6-98f9-59c39807d727.png)

**Comment by kotarohara** [comment]
Yeah, please look into `Point.render()` method.

**Comment by Soben713** [comment]
I fixed this problem for Firefox (pull request #321). Seems like if the original size of an SVG image is not specified in the file it could not be rendered in the canvas, which is why the curb-ramp icon was not shown in Firefox.

The problem seems to be extremely different in Safari. I haven't found out why it happens. It is not related to the filetype because I also tried to render PNG images and it didn't work either.

**Comment by jonfroehlich** [comment]
Did you find any Stackoverflow posts or related to the problem?

**Comment by Soben713** [comment]
@jonfroehlich Not really, earlier versions of Safari had some issues with drawing SVG images in canvas and most of the questions are related to it but it is now fixed in the current version. As I said it doesn't even work if we use PNG images, so it is not really related to that problem.

I'll get back to this issue after I finish resolving others.

**Comment by jonfroehlich** [comment]
Sounds good.

**Comment by jonfroehlich** [comment]
@Soben713, should we add Safari and Firefox support to the Public Deployment v2 milestone? I just checked on my Windows 10 computer and it seems like things work perfectly now in Safari 5.1.7 and work reasonably well in Firefox, except that the icons in placed labels don't show up--see screenshot.

![image](https://cloud.githubusercontent.com/assets/1621749/19420209/8da9b6f0-93b4-11e6-9c48-1d6d196647ee.png)

I just checked Google Analytics and almost 30% of our traffic is using Firefox or Safari (though perhaps half of the Safari traffic is coming from a touchscreen).

![image](https://cloud.githubusercontent.com/assets/1621749/19420143/8793f830-93b3-11e6-8788-6c0042d9d553.png)

In terms of devices, almost 23% of traffic is coming from a touchscreen.
![image](https://cloud.githubusercontent.com/assets/1621749/19420160/bc0b7ea8-93b3-11e6-93a5-5f982a09505d.png)

**Comment by Soben713** [comment]
@jonfroehlich Good catch about the "Other" labels' icons not showing up correctly. I know what the problem is and it will be an easy fix. And you are right, the main problem seems to be fixed in the new version of Safari. Although I still see a minor problem in Safari on desktop.

I am going to add this to Public Deployment v2.

**Comment by jonfroehlich** [comment]
We need to test on Safari and Firefox for Mac as well. I didn't do that in my limited time today.

Once we fix this, we can add 'Better support for Safari and Firefox' into our release notes.

**Comment by Soben713** [comment]
I tested them on my mac actually before my previous comment.

**Comment by Soben713** [comment]
TLDR;

I investigated this again and there were three problems. Two of them are fixed in PR #357 and the third one only occurs on our local servers. So we can close this issue.

----
### Safari/Firefox: Icons for 'Other' Labels Were Not Correctly Shown
This is now fixed:
![image](https://cloud.githubusercontent.com/assets/1576529/19454373/e7697d06-9486-11e6-9103-ec38cae82a74.png)

### Safari on Mac
Currently, if a label is added and then a label of another type is chosen, the cursor's icon in the GSV window shows the old icon on hover. This problem occurred because of a very strange bug in Safari and is now fixed.

### Safari: Local Server / Production Server 
Strangely the label icons are not shown on my local server but they are correctly shown in the deployed version. I guess Safari cannot render local images in the canvas (weird, but what else could be happening?). The related screenshots could be found below:

This is how it looks on my localhost (icons are missing):
![image](https://cloud.githubusercontent.com/assets/1576529/19454049/ad6f606c-9485-11e6-91f5-ed3f2903250f.png)

But the icons are correctly shown in the deployed version on the server:
![image](https://cloud.githubusercontent.com/assets/1576529/19454070/bbc4dd0e-9485-11e6-827e-d12a1bfef620.png)
